internal/engine/configs/configs_controller_test.go

```diff
@@ -73,11 +73,49 @@ func TestConfigsControllerDockerNotConnectedButNotRequired(t *testing.T) {
 	bar := manifestbuilder.New(f, "bar").
 		WithK8sYAML(testyaml.SanchoYAML).
 		Build()
-	a := f.run(bar)
+	_, a := f.run(bar)
 
 	assert.NoError(t, a.Err)
 }
 
+func TestBuildReasonTrigger(t *testing.T) {
+	f := newCCFixture(t)
+	defer f.TearDown()
+
+	f.addManifest("fe")
+	bar := manifestbuilder.New(f, "bar").WithK8sYAML(testyaml.SanchoYAML).Build()
+	f.setTiltfileHasBuilt()
+
+	state := f.st.LockMutableStateForTesting()
+	state.AppendToTriggerQueue(model.TiltfileManifestName, model.BuildReasonFlagTriggerWeb)
+	f.st.UnlockMutableState()
+
+	reloadStarted, _ := f.run(bar)
+
+	assert.Equal(t, model.BuildReasonFlagTriggerWeb, reloadStarted.Reason)
+}
+
+func TestBuildReasonTriggerAndOtherReason(t *testing.T) {
+	f := newCCFixture(t)
+	defer f.TearDown()
+
+	f.addManifest("fe")
+	bar := manifestbuilder.New(f, "bar").WithK8sYAML(testyaml.SanchoYAML).Build()
+	f.setTiltfileHasBuilt()
+
+	state := f.st.LockMutableStateForTesting()
+	state.AppendToTriggerQueue(model.TiltfileManifestName, model.BuildReasonFlagTriggerWeb)
+	state.PendingConfigFileChanges["somefile.txt"] = time.Now().Add(time.Second)
```

Comment:
this needs to be negative one second (in the past) not + 1 second (in the future)

Reply:
actually nm maybe i'm confused about what this is testing...

Reply:
yeah this wasn't a great way to write this test but i'm testing that the build reflects trigger AND another reason -- in this case that there were pending config changes. I'm adding a second to the timestamp here to make EXTRA SURE that this pending filechange is timestamped after the last tiltfile load (set in line 104). But let me see if there's not a better way to do this

Reply:
...honestly I'm not even sure this + "TestBuildReasonTrigger" are a useful set of tests. will simplify.